mobile/src/main/java/org/openhab/habdroid/ui/ViewExtensions.kt

```diff
@@ -111,3 +122,26 @@ fun RemoteViews.duplicate(): RemoteViews {
         clone()
     }
 }
+
+fun MaterialButton.setTextAndIcon(connection: Connection, mapping: LabeledValue) {
+    val iconUrl = mapping.icon?.toUrl(context, true)
+    if (iconUrl == null) {
+        icon = null
+        text = mapping.label
+        return
+    }
+    CoroutineScope(Dispatchers.IO + Job()).launch {
```

Comment:
Assign an intermediate drawable (loading indicator) here?

Reply:
Not sure. We don't do that for the icon of a widget/item as well and icons are/should be small, so fast to load and they are cached.